doc/rst/language/spec/generics.rst

```diff
@@ -790,6 +790,34 @@ instead.
    instance is ``IntegerTuple(3)``. The type specified by
    ``IntegerTuple`` is a generic type.
 
+
+   *Example (MemberWidth.chpl)*.
+
+   The following code defines a record called ``R`` that is generic over an
+   integer parameter ``width`` that describes the width of the integers
+   contained in the record's array field ``A``.
+
+   .. code-block:: chapel
+
+      record R {
+         param width: int;
+         var D = {1..10};
+         var A: [D] int(width);
+      }
+
+      config const big : bool;
+
+      if (big) {
+         var r = new R(64);
+         writeln(r.type:string);
+      } else {
+         var r = new R(32);
+         writeln(r.type:string);
+      }
+
+   Running this example with ``--sbig=true`` will print out ``R(64)``, and with
+   ``--sbig=false`` or no argument it will print out ``R(32)``.
```

Comment:
Looks good but can you get it to be generated as a test and tested? Use `make spectests` and see another spec example to see how to do it.